### ISSUE TITLE
[CoE Starter Kit - BUG] Admin | Audit Logs | Load events from exported Audit Log CSV file Loads Do Not Work

### ISSUE THREAD

**Issue by uzma-fis** [microsoft/coe-starter-kit#7131]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

Had an issue with collecting logs in COE as I didn't set up the API correctly. Admin | Audit Logs | Sync Audit Logs (V2) works as expected. 

I followed guidance https://learn.microsoft.com/en-us/power-platform/guidance/coe/setup-auditlog-http#how-to-get-older-data to get older data, I have reduced down the files down to smaller chunks (and even tried 1 row!) and I get an error. 

Action 'ParseAuditData' failed

Error occurs ParseAuditData. I have validate the raw and each output in JSON parser and cannot see issue. 

![image](https://github.com/microsoft/coe-starter-kit/assets/141040714/fb6a18e4-2ca0-4d90-b74a-0512fd0ac48d)

Can someone try to repo this? 

InvalidJSON. The 'content' property of actions of type 'ParseJson' must be valid JSON. The provided value cannot be parsed: 'Invalid character after parsing property name. Expected ':' but got: C. Path '', line 1, position 3.'.

### Expected Behavior

Flow should finish as expected

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

Aug 2023

### What app or flow are you having the issue with?

Admin | Audit Logs | Load events from exported Audit Log CSV file

### What method are you using to get inventory and telemetry?

Cloud flows

### Steps To Reproduce

Follow https://learn.microsoft.com/en-us/power-platform/guidance/coe/setup-auditlog-http#how-to-get-older-data step by step. 

### Anything else?

Used https://jsonlint.com/ to validate RAW input and was fine

AB#884

**Comment by Jenefer-Monroe** [comment]
Hoping one of the new crew can see if you repro this and if so, see if the instructions need updated.

**Comment by RahulBojji** [comment]
@uzma-fis Can you please share screenshots or json that is passed as input to the Parse "AuditData" action? Looks like some property in input might be null and because of this action is failing

**Comment by uzma-fis** [comment]
{
    "content": "{\"\"CreationTime\"\":\"\"2023-11-01T08:29:53\"\",\"\"Id\"\":\"\"00027fdf-05b1-4074-b59c-aa0b0ff8051a\"\",\"\"Operation\"\":\"\"LaunchPowerApp\"\",\"\"OrganizationId\"\":\"\"e3ff91d8-xxxx-xxxx-a0b4-18910a6ac575\"\",\"\"RecordType\"\":45,\"\"ResultStatus\"\":\"\"Success\"\",\"\"UserKey\"\":\"\"7369ac39-xxxx-4b12-b4ac-a4285ee62b75\"\",\"\"UserType\"\":0,\"\"Version\"\":1,\"\"Workload\"\":\"\"PowerApps\"\",\"\"ClientIP\"\":\"\"::ffff:00.0.0.00\"\",\"\"ObjectId\"\":\"\"7369ac39-fe8b-4b12-b4ac-a4285ee62b75\"\",\"\"UserId\"\":\"\".COM\"\",\"\"AdditionalInfo\"\":\"\"{\\\"\"environmentName\\\"\":\\\"\"f8ba516d-c54e-xxxx-b535-487f5a30i887c\\\"\"}\"\",\"\"AppName\"\":\"\"649fcbd9-8fd2-40a1-ae09-3e9c28d320a0\"\"}",
    "schema": {
        "type": "object",
        "properties": {
            "CreationTime": {
                "type": "string"
            },
            "Id": {
                "type": "string"
            },
            "Operation": {
                "type": "string"
            },
            "OrganizationId": {
                "type": "string"
            },
            "RecordType": {
                "type": "integer"
            },
            "ResultStatus": {
                "type": "string"
            },
            "UserKey": {
                "type": "string"
            },
            "UserType": {
                "type": "integer"
            },
            "Version": {
                "type": "integer"
            },
            "Workload": {
                "type": "string"
            },
            "ClientIP": {
                "type": "string"
            },
            "ObjectId": {
                "type": "string"
            },
            "UserId": {
                "type": "string"
            },
            "AdditionalInfo": {
                "type": "string"
            },
            "AppName": {
                "type": "string"
            }
        }
    }
}

**Comment by uzma-fis** [comment]
But there is no amendment from download from purview and loading CSV file. Can you repo on your end?

**Comment by Jenefer-Monroe** [comment]
I dont repro, no. This still works for us. You seem to have some extra quotes around your items as shown here:
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/072ad644-c7de-4de8-9de3-2f88105031af)

I'll share pictures of what I see along the way so we can see where you diverge. Do these look the same?

### DATA
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/2fca674a-e731-4da6-91b3-a0069aa36757)
```
"RecordId,CreationDate,RecordType,Operation,UserId,AuditData,AssociatedAdminUnits,AssociatedAdminUnitsNames\r",
  "a11a1111-a1a1-42bb-998b-20bd9c0ba8fb,11/16/2023 2:34:32 PM,45,LaunchPowerApp,Joe@company.onmicrosoft.com,\"{\"\"CreationTime\"\":\"\"2023-11-16T14:34:32\"\",\"\"Id\"\":\"\"a11a1111-a1a1-42bb-998b-20bd9c0ba8fb\"\",\"\"Operation\"\":\"\"LaunchPowerApp\"\",\"\"OrganizationId\"\":\"\"a11a1111-a1a1-aa11-a1a1-a1a1a1a1a1a1\"\",\"\"RecordType\"\":45,\"\"ResultStatus\"\":\"\"Success\"\",\"\"UserKey\"\":\"\"a11a1111-a1a1-4955-9276-3d9b30a67dd2\"\",\"\"UserType\"\":0,\"\"Version\"\":1,\"\"Workload\"\":\"\"PowerApps\"\",\"\"ClientIP\"\":\"\"::ffff:175.101.18.138\"\",\"\"ObjectId\"\":\"\"a11a1111-a1a1-4955-9276-3d9b30a67dd2\"\",\"\"UserId\"\":\"\"Joe@company.onmicrosoft.com\"\",\"\"AdditionalInfo\"\":\"\"{\\\"\"environmentName\\\"\":\\\"\"a11a1111-a1a1-ef6c-85fb-547fe1e64c15\\\"\",\\\"\"resourceDisplayName\\\"\":\\\"\"My app for doing things\\\"\",\\\"\"environmentDisplayName\\\"\":\\\"\"my-envt\\\"\"}\"\",\"\"AppName\"\":\"\"a11a1111-a1a1-4e93-9207-822f0a514d85\"\"}\",,\r",
<etc more rows>
```

### SELECT
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/9ce906bc-6c23-47a3-9110-15945dd4d7ea)
```
[
  {
    "AuditEntry": "a11a1111-a1a1-42bb-998b-20bd9c0ba8fb,11/16/2023 2:34:32 PM,45,LaunchPowerApp,Joe@company.onmicrosoft.com,\"{\"\"CreationTime\"\":\"\"2023-11-16T14:34:32\"\",\"\"Id\"\":\"\"a11a1111-a1a1-42bb-998b-20bd9c0ba8fb\"\",\"\"Operation\"\":\"\"LaunchPowerApp\"\",\"\"OrganizationId\"\":\"\"a11a1111-a1a1-aa11-a1a1-a1a1a1a1a1a1\"\",\"\"RecordType\"\":45,\"\"ResultStatus\"\":\"\"Success\"\",\"\"UserKey\"\":\"\"a11a1111-a1a1-4955-9276-3d9b30a67dd2\"\",\"\"UserType\"\":0,\"\"Version\"\":1,\"\"Workload\"\":\"\"PowerApps\"\",\"\"ClientIP\"\":\"\"::ffff:175.101.18.138\"\",\"\"ObjectId\"\":\"\"a11a1111-a1a1-4955-9276-3d9b30a67dd2\"\",\"\"UserId\"\":\"\"Joe@company.onmicrosoft.com\"\",\"\"AdditionalInfo\"\":\"\"{\\\"\"environmentName\\\"\":\\\"\"a11a1111-a1a1-ef6c-85fb-547fe1e64c15\\\"\",\\\"\"resourceDisplayName\\\"\":\\\"\"My app for doing things\\\"\",\\\"\"environmentDisplayName\\\"\":\\\"\"my-envt\\\"\"}\"\",\"\"AppName\"\":\"\"a11a1111-a1a1-4e93-9207-822f0a514d85\"\"}\",,\r"
  },
 <etc more rows>
]
```

**Comment by uzma-fis** [comment]
Thanks for the reproduction! I have figured it out and now its a mystery why there are double quotes. For anyone else just CTRL + H and find "" and replace with ".

I only imported a test line and its gone through fine

**Comment by Jenefer-Monroe** [comment]
Oh good. I'm glad we got you unblocked. I dont know how you ended in that state but seems like you have a good workaround. 
Please do let us know if the **_Admin | Audit Logs | Sync Audit Logs (V2)_** flow works for you. Likely it will and this was just an issue with the downloaded file but do let us know.

**Comment by Jenefer-Monroe** [comment]
closing out as no further action for starter kit team

**Comment by uzma-fis** [comment]
Hi guys so sorry for the late reply. I have had a nightmare trying to import the data. 

I had to split DOWN the data to 24,999k rows and ensure the header rows were accepted. 

I carried out testing as follows
-> audit log in csv with 1 row of data
-> ran the flow, success
-> using same file, insert 24,999k data
-> run flow - completes in 3 hours :) 

So the format of the file when we download form compliance/purview is not compatible with the flow import that is in the COE, some manipulation of the file is needed. 

Hope this helps others.
